packages/forms/src/UIForm/fields/NestedListView/NestedListView.component.js

```diff
@@ -0,0 +1,254 @@
+import PropTypes from 'prop-types';
+import React from 'react';
+import keycode from 'keycode';
+import ListView from '@talend/react-components/lib/ListView';
+import { translate } from 'react-i18next';
+
+import { I18N_DOMAIN_FORMS } from '../../../constants';
+import getDefaultT from '../../../translate';
+import { getItemsProps, initItems } from './NestedListView.utils';
+import FieldTemplate from '../FieldTemplate';
+
+import theme from './NestedListView.scss';
+
+const DISPLAY_MODE_DEFAULT = 'DISPLAY_MODE_DEFAULT';
+const DISPLAY_MODE_SEARCH = 'DISPLAY_MODE_SEARCH';
+const DEFAULT_ITEM_HEIGHT = 33;
+
+class NestedListViewWidget extends React.Component {
+	constructor(props) {
+		super(props);
+
+		const { t, value, schema, id } = props;
+
+		this.defaultHeaderActions = [
+			{
+				id: `${id}-search`,
+				icon: 'talend-search',
+				label: t('LISTVIEW_WIDGET_SEARCH', { defaultValue: 'Search for specific values' }),
+				onClick: this.switchToSearchMode.bind(this),
+			},
+		];
+
+		this.defaultSearchHeaderActions = [
+			{
+				id: 'abort',
+				icon: 'talend-cross',
+				label: t('LISTVIEW_WIDGET_ABORT', { defaultValue: 'Abort' }),
+				onClick: this.switchToDefaultMode.bind(this),
+			},
+		];
+
+		const toggledChildren = [];
+
+		this.state = {
+			...initItems(schema, value, null, toggledChildren, this.getInitItemsCallbacks()),
+			toggledChildren,
+			value,
```

Comment:
`value` don't seem to be used in render so should not be in state.

Reply:
It is used downwards in the .utils file's method

Reply:
But it's not used for the render so it's not a react state, it's just a class attribute 'internal state'. It will avoid checking for re-render.

Reply:
It is used to calculate the props of the rendered elements (displayedItems)

Reply:
@jsomsanith : Correct me if i'm wrong, but I think the react state must be used only for values used for the renders, not for the one used to calculate the values rendered.

It make things more simpler, example : in the current way, a lot's of internal base or pre-computed data are sent to the rendered components. => With default spreading props to the children in the majority of components, some values can easily get through the whole component tree for nothing unless adding potential side effects.
Even worse, some values like `items` are overridden between the state spreading and the custom defined props after ; which makes the code less declarative and more imperative.

Reply:
Not necessarily. It really represent the state of your component, and it can have internal state. It depends on the case.
Usually, you should not store things that can be calculated from props. But here it can be heavy, and the state is used for some kind of memoization. Nothing wrong IMO

What you say about items props that is overridden, it's more the state spreading that is wrong. We control all the state and know exactly what we want to pass. We should not spread it.